Bottom: b7be02c28b0a823e4febc6d0fe6280aca0572320
Top:    fb27d66af8d53d1e4651483b07d689795230c293
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 11:49:27 +0100

Expose draw.c internal drawing APIs


---

diff --git a/src/draw.c b/src/draw.c
index 10da1dd..c2e5c8c 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -100,10 +100,7 @@ static void SetPVColor (PinTypePtr, int);
 static void DrawEMark (ElementTypePtr, LocationType, LocationType, bool);
 static void ClearPad (PadTypePtr, bool);
 static void DrawHole (PinTypePtr);
-static void DrawMask (int side, BoxType *);
-static void DrawPaste (int side, BoxType *);
 static void DrawRats (BoxType *);
-static void DrawSilk (int side, const BoxType *);
 static int via_callback (const BoxType * b, void *cl);
 static int pin_callback (const BoxType * b, void *cl);
 static int pad_callback (const BoxType * b, void *cl);
@@ -576,7 +573,7 @@ clearPad_callback (const BoxType * b, void *cl)
  * Draws silk layer.
  */
 
-static void
+void
 DrawSilk (int side, const BoxType * drawn_area)
 {
 #if 0
@@ -635,7 +632,7 @@ DrawMaskBoardArea (int mask_type, BoxType *screen)
 /* ---------------------------------------------------------------------------
  * draws solder mask layer - this will cover nearly everything
  */
-static void
+void
 DrawMask (int side, BoxType * screen)
 {
   int thin = TEST_FLAG(THINDRAWFLAG, PCB) || TEST_FLAG(THINDRAWPOLYFLAG, PCB);
@@ -664,7 +661,7 @@ DrawMask (int side, BoxType * screen)
 /* ---------------------------------------------------------------------------
  * draws solder paste layer for a given side of the board
  */
-static void
+void
 DrawPaste (int side, BoxType *drawn_area)
 {
   gui->set_color (Output.fgGC, PCB->ElementColor);
diff --git a/src/draw.h b/src/draw.h
index c60b6d9..ff93f46 100644
--- a/src/draw.h
+++ b/src/draw.h
@@ -74,6 +74,10 @@ void EraseObject (int, void *, void *);
 void LoadBackgroundImage (char *);
 void UpdateAll (void);
 
+void DrawMask (int side, BoxType *);
+void DrawSilk (int side, const BoxType *);
+void DrawPaste (int side, BoxType *);
+
 /*GdkDrawable *draw_get_current_drawable(void);*/
 
 #endif
